### PR TITLE
feature/88-remove-all-the-old-inputs

### PR DESCRIPTION
**Pull request by bmirra01**
#88 

- updated the storybook with the new form components
- removed old inputs from the application
- use the new RHF inputs across Lumina
- Removed React-Forms-Component
- Removed the folder with the components created under that lib

### CODE REVIEW

**Review by AdrianaAC — APPROVED** [review]

**Comment by bmirra01** [comment]
@JWebCoder check the latest commit

File: packages/lumina-core/src/components/form-components/input/index.tsx [hunk 1]
```diff
@@ -1,12 +1,32 @@
-import { Input, type ControlLogicProps } from 'react-form-component'
+import * as React from 'react'
```
Comment (JWebCoder) [inline packages/lumina-core/src/components/form-components/input/index.tsx:1]: Do not import React, but import only the need section

File: packages/lumina-core/src/components/form-components/utils/hooks/useActivateOnEnter/index.tsx [hunk 2]
```diff
@@ -21,38 +23,39 @@ export const useActivateOnEnter = ({
...
-      } as ChangeEvent<HTMLInputElement>;
-      onChange(event);
+      } as ChangeEvent<HTMLInputElement>
+      onChange(event)
     }
-  }, [value, onChange]);
+  }, [value, onChange])
 
   const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    console.log(handleKeyDown)
```
Comment (JWebCoder) [inline packages/lumina-core/src/components/form-components/utils/hooks/useActivateOnEnter/index.tsx:40]: Remove the logs

File: packages/lumina-docs/src/stories/componentsLumina/forms/hooks/UseActivateOnEnter.mdx [hunk 3]
```diff
@@ -0,0 +1,111 @@
+import { Meta, ColorPalette, ColorItem } from '@storybook/blocks'
+
+<Meta title='Components Lumina/Forms/Hooks/useActivateOnEnter/useActivateOnEnter' />
+
+# useActivateOnEnter Custom Hook
+
+## Concepts
+
+The main purpose of this hook is to extend a component to have the possibility to the user to submit something with a enter on press, specifically while using `<Input />` component.
```
Comment (JWebCoder) [inline packages/lumina-docs/src/stories/componentsLumina/forms/hooks/UseActivateOnEnter.mdx:9]: "for the user" and "something by pressing enter"

**Review by JWebCoder — CHANGES_REQUESTED** [review]

File: packages/lumina-core/src/components/form-components/input/index.tsx (hunk 1 above)
Comment (bmirra01) [reply packages/lumina-core/src/components/form-components/input/index.tsx:1]: Sorted

File: packages/lumina-core/src/components/form-components/utils/hooks/useActivateOnEnter/index.tsx (hunk 2 above)
Comment (bmirra01) [reply packages/lumina-core/src/components/form-components/utils/hooks/useActivateOnEnter/index.tsx:40]: sorted

File: packages/lumina-docs/src/stories/componentsLumina/forms/hooks/UseActivateOnEnter.mdx (hunk 3 above)
Comment (bmirra01) [reply packages/lumina-docs/src/stories/componentsLumina/forms/hooks/UseActivateOnEnter.mdx:9]: done

File: packages/lumina-core/src/components/form-components/input/index.tsx [hunk 4]
```diff
@@ -1,12 +1,32 @@
-import { Input, type ControlLogicProps } from 'react-form-component'
+import React from 'react'
```
Comment (JWebCoder) [inline packages/lumina-core/src/components/form-components/input/index.tsx:1]: Do not import React, import only what is needed.

**Comment by bmirra01** [comment]
ready for review/merge @JWebCoder

**Review by JWebCoder — CHANGES_REQUESTED** [review]

File: packages/lumina-core/src/components/form-components/input/index.tsx (hunk 4 above)
Comment (bmirra01) [reply packages/lumina-core/src/components/form-components/input/index.tsx]: fixed

File: packages/lumina-core/src/components/form-components/utils/utility-components/label/index.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-interface ILabelProps extends InputHTMLAttributes<HTMLLabelElement> {
+type TLabelProps = InputHTMLAttributes<HTMLLabelElement> & {
   className?: string
   label?: string
 }
 
-export const LabelTitle = forwardRef<HTMLLabelElement, ILabelProps>(
+export const LabelTitle = forwardRef<HTMLLabelElement, TLabelProps>(
   ({ className, label, ...rest }, ref) => {
     return (
```
Comment (JWebCoder) [inline packages/lumina-core/src/components/form-components/utils/utility-components/label/index.tsx:11]: if there is no label text, then return null

File: packages/lumina-core/src/components/form-components/utils/utility-components/help/index.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-interface IHelpTextProps extends HTMLAttributes<HTMLParagraphElement> {
+type THelpTextProps = HTMLAttributes<HTMLParagraphElement> & {
   className?: string
   help?: string
 }
 
-export const HelpText = forwardRef<HTMLParagraphElement, IHelpTextProps>(
+export const HelpText = forwardRef<HTMLParagraphElement, THelpTextProps>(
   ({ className, help, ...rest }, ref) => {
     return (
```
Comment (JWebCoder) [inline packages/lumina-core/src/components/form-components/utils/utility-components/help/index.tsx:11]: if there is no help text, return null

**Review by JWebCoder — CHANGES_REQUESTED** [review]

**Comment by bmirra01** [comment]
@JWebCoder Should be done now

**Review by JWebCoder — APPROVED** [review]

**Pull request merged**